pilot/pkg/model/config.go

```diff
@@ -50,33 +51,26 @@ func (key NamespacedName) String() string {
 // ConfigKey describe a specific config item.
 // In most cases, the name is the config's name. However, for ServiceEntry it is service's FQDN.
 type ConfigKey struct {
-	Kind      config.GroupVersionKind
+	Kind      config.Kind
 	Name      string
 	Namespace string
 }
 
 func (key ConfigKey) HashCode() uint64 {
 	hash := md5.New()
-	for _, v := range []string{
-		key.Name,
-		key.Namespace,
-		key.Kind.Kind,
-		key.Kind.Group,
-		key.Kind.Version,
-	} {
-		hash.Write([]byte(v))
-	}
-	var tmp [md5.Size]byte
-	sum := hash.Sum(tmp[:0])
+	hash.Write([]byte(kind.String(key.Kind)))
+	hash.Write([]byte(key.Name))
+	hash.Write([]byte(key.Namespace))
+	sum := hash.Sum(nil)
 	return binary.BigEndian.Uint64(sum)
 }
 
 func (key ConfigKey) String() string {
-	return key.Kind.Kind + "/" + key.Namespace + "/" + key.Name
+	return kind.String(key.Kind) + "/" + key.Namespace + "/" + key.Name
```

Comment:
I would make a .String method on `Kind` instead of having a standalone function

Reply:
yup, i could add another file in the same package as the generated file with only the declaration, does that seem ok?

I was aiming at leaving the declaration in `config` just to be consistent with `GroupVersionKind`.

Reply:
yeah I think it would be useful to have the .String(); it makes it so when you print it the string will automatically be used as well